src/modification_api.jl

```diff
@@ -266,23 +262,262 @@ function set_vector_related!(
     source_element[relation_field] = Int[]
     for target_index in target_indices
         target_element = _get_element(data, target, target_index)
-        push!(
-            source_element[relation_field],
-            target_element["reference_id"],
-        )
+        push!(source_element[relation_field], target_element["reference_id"])
     end
 
     return nothing
 end
 
 function Base.show(io::IO, data::Data)
-    collections = get_collections(data)
+    summary(io, data)
+end
+
+function create_study(
+    ::OpenInterface;
+    data_path::AbstractString = pwd(),
+    pmd_files::Vector{String} = String[],
+    pmds_path::AbstractString = PMD._PMDS_BASE_PATH,
+    netplan::Bool = false,
+    kws...,
+)
+    if !isdir(data_path)
+        error("data_path = '$data_path' must be a directory")
+    end
+
+    json_path = joinpath(data_path, "psrclasses.json")
+
+    # Select mapping
+    model_class_map = if netplan
+        PMD._MODEL_TO_CLASS_NETPLAN
+    else
+        PMD._MODEL_TO_CLASS_SDDP
+    end
 
-    print(
-        io,
-        """
-        PSRClasses Interface Data with $(length(collections)) collections:
-            $(join(collections, "\n    "))
-        """
+    data_struct, model_files_added = PMD.load_model(pmds_path, pmd_files, model_class_map)
+
+    data = Data(
+        raw = Dict{String,Any}(),
+        data_path = data_path,
+        data_struct = data_struct,
+        validate_attributes = false,
+        model_files_added = model_files_added,
+        stage_type = STAGE_WEEK,
+        first_year = 2023,
+        first_stage = 1,
+        first_date = Dates.Date(2023, 1, 1),
+        controller_date = Dates.Date(2023, 1, 1),
+        duration_mode = FIXED_DURATION,
+        number_blocks = 1,
+        log_file = nothing,
+        verbose = true,
     )
+
+    create_element!(data, "PSRStudy")
+
+    return data
+end
+
+function _validate_collection(data::Data, collection::String)
+    data_struct = get_data_struct(data)
+
+    if !haskey(data_struct, collection)
+        error("Collection '$collection' is not available for this study")
+    end
+
+    return nothing
+end
+
+function _validate_attribute(::Data, ::String, attribute::String, ::T) where {T}
+    return error("Invalid type '$T' assigned to attribute '$attribute'")
+end
+
+function _validate_attribute(
+    data::Data,
+    collection::String,
+    attribute::String,
+    value::Vector{T},
+) where {T<:MainTypes}
+    attribute_struct = get_attribute_struct(data, collection, attribute)
+
+    if !attribute_struct.is_vector
+        error("Vectorial value '$value' assigned to scalar attribute '$attribute'")
+    end
+
+    if !(T <: attribute_struct.type)
+        error(
+            "Invalid element type '$T' for attribute '$attribute' of type '$(attribute_struct.type)'",
```

Comment:
Dont we have those _check_xxx methods that we use in the getters?

Reply:
Having `_check_*()::Bool` methods is a good idea to encapsulate multiple validation steps.

I'm not very happy with throwing errors from inside them because we sometimes lose information and specificity when writing error messages. In some cases we also need to pass extra parameters to `_check_*` just for them to compose te message.